src/main/resources/software/amazon/smithy/lsp/language/build.smithy

```diff
@@ -0,0 +1,90 @@
+$version: "2.0"
+
+namespace smithy.lang.server
+
+structure SmithyProjectJson {
+    sources: Strings
+    imports: Strings
+    outputDirectory: String
+    dependencies: ProjectDependencies
+}
+
+list ProjectDependencies {
+    member: ProjectDependency
+}
+
+structure ProjectDependency {
+    name: String
+
+    @required
+    path: String
+}
+
+structure SmithyBuildJson {
+    @required
+    version: SmithyBuildVersion
+
+    outputDirectory: String
+    sources: Strings
+    imports: Strings
+    projections: Projections
+    plugins: Plugins
+    ignoreMissingPlugins: Boolean
+    maven: Maven
+}
+
+@default("1")
+string SmithyBuildVersion
+
+map Projections {
+    key: String
+    value: Projection
+}
+
+structure Projection {
+    abstract: Boolean
+    imports: Strings
+    transforms: Transforms
+    plugins: Plugins
+}
+
+map Plugins {
+    key: String
+    value: Document
+}
+
+list Transforms {
+    member: Transform
+}
+
+structure Transform {
+    @required
+    name: String
+
+    args: TransformArgs
+}
+
+structure TransformArgs {
+}
+
+structure Maven {
+    dependencies: Strings
+    repositories: MavenRepositories
+}
+
+list MavenRepositories {
+    member: MavenRepository
+}
+
+structure MavenRepository {
+    @required
```

Comment:
How are constraints used by the LS? Could we possibly add more constraints to save energy doing validations like urls etc

Reply:
They really aren't right now. Originally I intended to use them for validation, but I scrapped that idea to use the validation errors from smithy-build. However, with trait values, required members get autopopulated in completions, so maybe a future update could do that for node values too.